### ISSUE TITLE
aggr, aggr-server and SignificantTrades (?)

### ISSUE THREAD

**Issue by carloswm85** [Tucsky/aggr#1]
Hello, there. 

I just wanted to know what's the relationship between this three project (?). I'm currently using SignificantTrades, and I can appreciate the work you've done. Thank you.

Also, I may ask you what are your future plans for this.

**Comment by Tucsky** [comment]
[Aggr](https://github.com/Tucsky/aggr) and [SignificantTrades](https://tucsky.github.io/SignificantTrades) are the same, just different versions! I made another repo and archived SignificantTrades because it was still that fork of a project that has very little to do with this current app at this point.

[aggr-server](https://github.com/Tucsky/aggr-server) is just the backend part used for collecting / serving historical data 

https://tucsky.github.io/SignificantTrades is v1
https://aggr.trade/ is currently v2
this repo is v3 (beta.aggr.trade)

**Comment by carloswm85** [comment]
Ok. I've been programming for the last 2 years, so I'm still an apprentice on all of this. Expect some dumb questions.

Are you currently working on v2 or v3? I have some suggestions to make to it, by the way. Let me know if you're interested in. I'll be forking v1 or v2.

Question: Did you deploy the app using GitHub Pages? [https://cli.vuejs.org/guide/deployment.html#github-pages](url)

I don't know if the issues tab is the best place to just talk about this, what about GitHub discussions? [https://docs.github.com/en/discussions/quickstart](url)

**Comment by Tucsky** [comment]
> Question: Did you deploy the app using GitHub Pages

no but this is a good idea!


> Are you currently working on v2 or v3

Working hard on v3, v2 that is currently served on https://aggr.trade/ will be replaced very soon ™️.


> And why does the image looks so different than the actual web apps? Image That seems to me to be the case.

This picture 2.5 year old! lol
I will make a new one for v3 👍🏼

**Comment by carloswm85** [comment]
Ok. That's great. I have some other questions/suggestions.

1) What style of trading do you practice? I've seen in your app there are a lot of orders. Are those order flows? I don't know about that topic. I trade using Wyckoff Schematics, and Volume Spread Analysis, mainly, and some price action. Here's my profile at [here](https://www.tradingview.com/u/BlueJayBird/#published-scripts) As you'll see, I've published some indicators on volume.

I can invite more people to use your app if you want.

Now, I'd like to talk about the app.

I've used the agg.trade page on my Android TV. And I have some suggestions to it.

![WhatsApp Image 2021-05-26 at 10 54 21 PM](https://user-images.githubusercontent.com/29048235/119754977-a1362d00-be77-11eb-99bb-f6befad28f2e.jpeg)

2) It's really hard to follow price and volume vertically (and horizontally as well). It would be great if you'd added the gridlines. As you can see I've done that little modification to my local copy, but I've hard coded the change, and I've not added the setting options to the app. As I told you before, I'm new at this, and I don't know Vue nor any other JS framework.

Take a closer look and you'll see the thin line there.

![Screenshot 2021-05-26 230824](https://user-images.githubusercontent.com/29048235/119755128-e2c6d800-be77-11eb-9cc8-45d932dd67ee.png)

By the way, I installed and used your app in a Google Chrome apk installation in the Android TV, which you see in the first picture. As you can see there, volume is not visible, and not all the screen is used. I used another app for seen the app correctly there, which is called, [Open Browser, but for Android TV](https://play.google.com/store/apps/details?id=com.seraphic.openinet.pre&hl=en_US&gl=US)

![2](https://user-images.githubusercontent.com/29048235/119756149-82d13100-be79-11eb-9619-3d187ec0c8fa.jpeg)

3) Consider please adding one button/click options for full-screen, and for hiding the side bar with the buys and sell.

I don't want to be too demanding on my requests, so that's it.

As a final note, I'm willing to collaborate in the future, as long as my free time and desires to trade allow it.

**Comment by Tucsky** [comment]
Thanks a lot for all suggestions my friend! 
Yes of course I can add an option for gridlines. I don't like gridlines them too much tbh but can understand the need.

> What style of trading do you practice?
 
Mostly Fibs & VPVR. This app is more to get a sense of the market, get to see how the market react at specific levels that I find on my personal trading tool. I'm addicted to the audio feature. For the disclaimer I'm more of a dev than a trader!! :) 

> By the way, I installed and used your app in a Google Chrome apk installation in the Android TV, which you see in the first picture. As you can see there, volume is not visible, and not all the screen is used. I used another app for seen the app correctly there, which is called, Open Browser, but for Android TV

Not sure what is the issue here, I see volume on your screenshot. Are you saying there is a bug hiding volume and sidebar ? Btw this is very specific to v2 as **there is no sidebar anymore** on v3 just modules (called "panes", trades pane, chart, stat, counter, market) that you can add / remove, assign markets to them and move arround to your liking.
Ive deployed to github http://tucsky.github.io/aggr

> Consider please adding one button/click options for full-screen, and for hiding the side bar with the buys and sell

Yes good idea. And still very relevant in v3 to full screen one module on top of the others.

> I don't want to be too demanding on my requests, so that's it.

Thanks you :)

> As a final note, I'm willing to collaborate in the future, as long as my free time and desires to trade allow it.

Contributing is appreciated and encouraged ! This is a free time project for me and time goes super fast most of the days. If you (or anyone) willing to contribute have questions on how I would go with implementing this or that I can help of course. The transition to typescript since v3 should makes things easier I hope.

**Comment by carloswm85** [comment]
> Not sure what is the issue here, I see volume on your screenshot. 

Not really an issue you should fix, I think. In the APK Chrome, the volume is not visible, because the search bar is pushing the app down. But in the Open Browser app, volume is indeed visible. I would say it's just a Chrome issue that may be fix with your addition of the full-screen option.

> Ive deployed to github http://tucsky.github.io/aggr

Fantastic!

For now, I'm not interested in the other panes, other than chart's. I have some suggestion about them:

- When the app is started there are 3 panes on the left. If I remove them, the chart pane does not expand to occupy the available space.
- When any pane is added from the upper bar (where the settings are, I mean), they've weirdly added to the bottom of the viewport, stealing space from the chart pane.
- I liked the price pane the most below the chart as in V1. ;)

> The transition to typescript since v3 should makes things easier I hope.

Ok. Probably I'll have to learn about that too, but I'm still just a rookie at development.

I'll download the current v3, and work with it. Can I add and push indicators to it? I would like to add some EMAs (3 of them, in fact; 20/50/200).

I have a very specific question. What's the meaning of the gray volume bars behind the volume bars? Do you have any documentation about how to read the other color bars? In v3. 

I think it would be great if you open a discussion section at this repo. I think issues and discussions are different concerns. Repo > Settings > Features > Discussions

**Comment by carloswm85** [comment]
@Tucsky Thanks a lot man for the improvements you've made. I have no words to thank you for the work you've done. Maybe some $$$ collaboration in the future for the project when I become really profitable.

Thanks specially for the EMAs and the gridlines. They make reading the price on my 50" Smart TV way easier. The input format information in the EMAs is a great feature, by the way. Do you think you can add an input field for switching between low, high, open, close, hl2, hlc3, ohlc4?

I have some other questions:

1. Have you thought about adding a discussion section to this repo as I suggested?
2. Have you considered making documentation to the repo about the instruments available in the chart? I don't understand some of the volume columns, the sells/buys panes, and others.
3. Also, I'm not sure about sharing the link to the GitHub Page deployment, since, as far as I know, the amount of calls that can be done to their servers is limited. And maybe that limit the access we have to the page. Do you know something about it?

Thanks again for your work.

PS. I'm taking some classes on Vue on my free time. Any suggestion over this?

**Comment by Tucsky** [comment]
Thanks man appreciate the feedbacks as always ! The app is shaping up nicely indeed !

> Have you thought about adding a discussion section to this repo as I suggested?

I don't really see the need right now tbh, not very crowded out there might as well continue the discussion here

> Have you considered making documentation

There is a little bit of documentation included about how to write scripts (help button in the indicator dialog window), it might give people a clue about what is going in the built-in indicators as well (what vbuy, lbuy, cbuy stand for etc...)
So about writing a wik,i I probably won't any time soon, I'd say PR is always encouraged, I can answers question about the features here.  All feature here are pretty basics, this is all aggregations of data reported by exchange with none to very minor calculations. 

> as far as I know, the amount of calls that can be done to their servers is limited

Oh is that true ? Have you noticed downtime on http://tucsky.github.io/aggr ? I didn't know there was a limit but most of the users won't use the github url anyway 
I'm will route some of the user base to the beta next week see how it goes, and taking down the server part behind https://tucsky.github.io/SignificantTrades so http://tucsky.github.io/aggr can use it, current one is a basic EC2 with very poor performance and high cost, I was just testing the service, not convinced at all lol

> I'm taking some classes on Vue on my free time

Well that's fantastic! Love vuejs 👍🏼 
If you are just getting into vue might as well start with vue3, which is quite different from vue2, they say its smaller, faster, and more powerful; I believe many big vuejs libraries will start to support it by the end of this year
Started this project as a way to learn the framework, so that was pretty much my course, it's in vue2 but yeah vue3 is def in the pipes

**Comment by Tucsky** [comment]
Closing this for now, I might open discussion section but gotta clean up the list :)

**Comment by carloswm85** [comment]
@Tucsky Okay. What list are you talking about?

And what's PR?

Thanks for your time. I'll be following up in the discussion board. I've been pretty inactive in the trading side of things, because of college. I really have little spare time for following up the charts in mid of the semester.

**Comment by carloswm85** [comment]
@Tucsky I wonder if you'd find this interesting: [http://3.129.240.52/](http://3.129.240.52/) The link is safe, as far as I know. It is from one guy in this Facebook group, HIS group: [https://www.facebook.com/groups/beatdealer](https://www.facebook.com/groups/beatdealer) He explains the basics of this in the group. It make sense to me, though I don't know if it is possible to use it in Bitcoin.
